src/cli/parse-flags.ts

```diff
@@ -42,14 +42,6 @@ export const parseFlags = (args: string[]): ConfigFlags => {
     }
   }
 
-  // to find unknown / unrecognized arguments we filter `args`, including only
-  // arguments whose normalized form is not found in `knownArgs`. `knownArgs`
-  // is populated during the call to `parseArgs` above. For arguments like
-  // `--foobar` the string `"--foobar"` will be added, while for more
-  // complicated arguments like `--bizBoz=bop` or `--bizBoz bop` just the
-  // string `"--bizBoz"` will be added.
-  flags.unknownArgs = flags.args.filter((arg: string) => !flags.knownArgs.includes(parseEqualsArg(arg)[0]));
```

Comment:
I decided to remove this filter here and to instead proactively build up the `unknownArgs` array as we're parsing. You'll see below that when we get into the `else` clauses of both `parseCLITerm` and `setCLIArg` there is new code to stick any argument which we're not handling into `flags.unknownArgs`.

Many tests cases which look for the presence or absence of values in this array continue to pass with these changes, so I believe they are correct to do.

But why do them?

Well, take a look at this filtering code. In `main` right now we're doing:

```ts
flags.unknownArgs = flags.args.filter((arg: string) => !flags.knownArgs.includes(parseEqualsArg(arg)[0]));
```

Basically, we declare the `unknownArgs` to be any arguments in the `args` where a 'normalized' version of the arg as calculated by `parseEqualsArg` isn't found in `knownArgs`.

This breaks down for handling aliased Jest args, in particular because I found that in order to reliably pass these arguments to Jest it is best to pass the 'desugared'[^1] form of the alias to Jest (through the `knownArgs`) rather than just the alias.

So in order to get that to work we want to make sure that the argument we're pushing onto `knownArgs` is full argument name, but this means that the normalized form of the _raw argument_ (i.e. the flag as set at the CLI by the user) will no longer be present in the `knownArgs` array, so this filter isn't reliable anymore.

As an example, if I do

```sh
npx stencil test --spec -u
```

that will result in a `knownArgs` that looks like

```ts
[ "--unknownArgs" ]
```

rather than

```ts
[ "-u" ]
```

Please let me know if that doesn't make sense, it's a bit hard to explain well!

[^1]: My  overloaded term for the full config flag name that an alias refers to, so e.g. `'-c'` will desugar to `'--config'`, `'-u'` to `'--updateSnapshot'`, etc. Also by 'best' I mean that without desugaring these aliases there are difficult-to-pin-down circumstances under which they are picked up by Jest _and_ circumstances under which they _aren't_. I stopped investigating once I realized that passing the desugared form always works.